libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
Sort of, those on-demand workers "should" have been working on on-demand requests in this cycle, but a prefetch worker did it instead.

I don't follow how we introduce extra CPU/network usage (other than to flush the extra requests).